allenact/utils/experiment_utils.py

```diff
@@ -1020,12 +1022,20 @@ def load_state_dict(self, state_dict: Dict[str, Any]):
                 "Loaded state dict was saved using an older version of AllenAct."
                 " If you are attempting to restart training for a model that had an off-policy component, be aware"
                 " that logging for the off-policy component will not behave as it previously did."
+                " Additionally, while the total step count will remain accurate, step counts"
+                " associated with losses will be reset to step 0."
             )
 
         for ps, stage_info in zip(self.pipeline_stages, state_dict["stage_info_list"]):
             ps.early_stopping_criterion_met = stage_info["early_stopping_criterion_met"]
             ps.steps_taken_in_stage = stage_info["steps_taken_in_stage"]
 
+            if "storage_uuid_to_steps_taken_in_stage" in stage_info:
```

Comment:
Haha thanks :) basically what's happening now is that each of the different types of storage track how many "steps" they have taken (e.g. the amount of offpolicy storage steps might be different than the number of onpolicy steps). As each storage gets its own tensorboard section, this makes it possible to track the training loss metrics corresponding to that storage using it's definition of "steps." The above saves each storages' step count so that these can be correctly displayed in the tensorboard when restarting a training pipeline.

Reply:
👍